server/SwipeDecision.py

```diff
@@ -15,10 +16,58 @@ def getPotentialMatchList(currentUserId):
             cursor.execute(PotentialMatchQuery, userID)
             potentialListId = [i[0] for i in cursor.fetchall()]
 
+            newPotentialMatchList = []
+
+            if(len(potentialListId) != 0):
+
+                findLocation = "SELECT longitude, latitude FROM Users WHERE id = %s"
+                cursor.execute(findLocation, userID)
+                currentUserLocation = cursor.fetchall()
+
+                if(currentUserLocation[0][0] != None):
+                    for i in currentUserLocation:
+                        currentUserLongitude = float(i[0])
+                        currentUserLatitude = float(i[1])
+
+                    for pmatch in range(len(potentialListId)):
+                        ID = (potentialListId[pmatch],)
+                        cursor.execute(findLocation, ID)
+                        potentialMatchLocation = cursor.fetchall()  
+
+                        if(potentialMatchLocation[0][0] != None):
+
+                            for i in potentialMatchLocation:
+                                potentialMatchLongitude = float(i[0])
+                                potentialMatchLatitude = float(i[1])
+                                
+                            p = pi/180
+                            a = 0.5 - cos((currentUserLatitude-potentialMatchLatitude)*p)/2 + cos(potentialMatchLatitude*p) * cos(currentUserLatitude*p) * (1-cos((currentUserLongitude-potentialMatchLongitude)*p))/2
+                            distance = 12742 * asin(sqrt(a)) 
+
+                            findMaxDistance = "SELECT maxDistance FROM Profile WHERE userId = %s"
```

Comment:
when you look for the current user longitude and latitude you should get their max distance and store that somewhere as well. Try to minimize the number of database calls